src/packet/literal_data.rs

```diff
@@ -940,19 +931,19 @@ mod tests {
 
             let string = random_string(&mut rng, len);
 
-            if !string.contains('\n') && !string.contains('\r') {
+            // We want to produce strings with "illegal" linebreaks.
+            // So we skip all strings that don't fit this goal.
+            //
+            // Test strings for this case need to contain a LF.
+            // And they must not normalize to legal CR+LF (like CR+CR+LF) does.
```

Comment:
I don't understand this comment.  CR+CR+LF might be the most contentious part of normalization, as it distinguishes the implementers who treat `\r?\n` as the line ending from those who treat `\r*\n` as the line ending.